investos/portfolio/risk_model/factor_risk.py

```diff
@@ -1,23 +1,351 @@
+        ]
+
+        print("Done pulling data")
+        return
+
+    def _generate_risk_models(self, timestamp="datetime") -> None:
+        """fit a cross sectional factor model and generate factor return covariance and idio returns variance
 
-        self._drop_excluded_assets()
+        Parameters
+        ----------
+        self._df_loadings : DataFrame
+            Linear regression model fits a linear model
+        self._fos_risk_factor_uuids : list
+            the array of X cols (AKA factor columns, or training data columns)
+        self._regress_col : string
+            the y col, by default return_1d
+
+        Returns
+        -------
+        """
+
+        """
+        fit the cross sectional regression
+        """
+        print("\n2. Running Cross Sectional Regression...")
+        self._fos_industry_factors = [
+            col
+            for col in self._df_loadings.columns
+            if col
+            not in ["datetime", "id"]
+            + self._fos_risk_factor_uuids
+            + [self._fos_return_uuid]
+        ]
+        scaler = StandardScaler()
+        self._factor_returns = {}
+        for date in sorted(self._df_loadings[timestamp].unique()):
+            # isolating the data for the current date
+            df_current = self._df_loadings[self._df_loadings[timestamp] == date]
+
+            # winsorzie and standarize risk factors
+            for col in self._fos_risk_factor_uuids:
+                df_current.loc[:, col] = mstats.winsorize(
+                    df_current.loc[:, col], limits=(0.05, 0.95)
+                )
+                df_current.loc[:, col] = scaler.fit_transform(
+                    df_current.loc[:, col].values.reshape(-1, 1)
+                ).flatten()
+            # winsorzie return
+            df_current.loc[:, self._fos_return_uuid] = mstats.winsorize(
+                df_current.loc[:, self._fos_return_uuid], limits=(0.05, 0.95)
+            )
+
+            y, X = (
+                df_current.loc[:, self._fos_return_uuid],
+                df_current.loc[
+                    :, (self._fos_risk_factor_uuids + self._fos_industry_factors)
+                ],
+            )
+            X.loc[:, "const"] = (
+                1  # need the intercept column, will convert this to country factor later
+            )
+            model = sm.OLS(y, X)
+            results = model.fit()
+            # store coefficients, intercept, feature_names, r2, and t-values into self._factor_returns based on the date
+            self._factor_returns[date] = {
+                "coefficients": results.params,
+                "intercept": results.params[
+                    "const"
+                ],  # the intercept needs the X col of all 1
+                "feature_names": self._fos_risk_factor_uuids
+                + self._fos_industry_factors,
+                "r2": results.rsquared_adj,
+                "t-values": results.tvalues,
```

Comment:
No dashes in key names. Only underscores. I think I left this comment last time.